### PR TITLE
Option for systemd-timesyncd

### PR DESCRIPTION
**Pull request by builderjer**
requires PR from `ovos-PHAL-plugin-system`
https://github.com/OpenVoiceOS/ovos-PHAL-plugin-system/pull/21

### CODE REVIEW

File: ovos_utils/system.py [hunk 1]
```diff
@@ -56,6 +56,15 @@ def ntp_sync():
     subprocess.call('ntpd -gq', shell=True)
     subprocess.call('service ntp start', shell=True)
 
+def timesync_sync(sudo=True):
```
Comment (NeonDaniel) [inline ovos_utils/system.py:59]: Can this default to `sudo=False` to match other methods here? Also, please add a docstring like other methods

Comment (NeonDaniel) [inline ovos_utils/system.py:59]: Please add unit test coverage; shutdown or reboot tests can be mostly copy/pasted with a minor change to the expected subprocess call

**Review by NeonDaniel — CHANGES_REQUESTED** [review]

File: ovos_utils/system.py (hunk 1 above)
Comment (builderjer) [reply ovos_utils/system.py:59]: I don't think defaulting to False would work.  The systemctl command needs sudo to stop and start.  Others are defaulting to True also
https://github.com/OpenVoiceOS/ovos-utils/blob/45b87776ae4b2556f8db502685b3d49b6b73295e/ovos_utils/system.py#L60
and others also.

I will add a docstring for this and `ntp_sync()`

**Comment by builderjer** [comment]
I deprecated all of the methods that I felt should go into the `ovos-PHAL-plugin-system` and made it an Admin Plugin

File: ovos_utils/system.py [hunk 2]
```diff
@@ -24,6 +24,72 @@ class MycroftRootLocations(str, Enum):
...
+# system utils
+@deprecated("DEPRECATED: use ovos-PHAL-plugin-system", "0.1.0")
+def ntp_sync():
+    # """
+    # Force the system clock to synchronize with internet time servers
+    # """
+    # subprocess.call('service ntp stop', shell=True)
+    # subprocess.call('ntpd -gq', shell=True)
+    # subprocess.call('service ntp start', shell=True)
+    self.bus.emit(Message("system.ntp.sync"))
```
Comment (JarbasAl) [inline ovos_utils/system.py:36]: self.bus does not exist, you need to create a new connection from scratch. maybe add bus=none to the arguments and then bus = bus or get_mycroft_bus()

File: test/unittests/test_system.py [hunk 3]
```diff
@@ -8,9 +8,23 @@ def test_is_running_from_module(self):
...
+
+    @patch("subprocess.Popen")
+    def test_timesync_sync(self):
+        from ovos_utils.system import timesync_sync
+        timesync_sync()
+        popen.assert_called_with("sudo systemctl stop systemd-timesyncd", shell=True)
+        popen.assert_called_with("sudo systemctl start systemd-timesyncd", shell=True)
+        timesync_sync(False)
+        popen.assert_called_with("systemctl stop systemd-timesyncd", shell=True)
+        popen.assert_called_with("systemctl start systemd-timesyncd", shell=True)
```
Comment (JarbasAl) [inline test/unittests/test_system.py:27]: these tests should move to the PHAL plugin, in here we can test that the correct bus message is emitted

**Comment by j1nx** [comment]
Can someone help me with rebasing this?

am in process of getting the PHAL services into shape. The old ntpd not found bug is one of the issues to resolve.

**Comment by builderjer** [comment]
I remember now.  There is no `self.bus` so all of that needed swapped out.  I will work on this this weekend

**Review by JarbasAl — APPROVED** [review]

**Pull request merged**